ktor-utils/src/io/ktor/util/pipeline/PipelineContext.kt

```diff
@@ -97,13 +102,16 @@ private class SuspendFunctionGun<TSubject : Any, TContext : Any>(
         private set
 
     private var rootContinuation: Any? = null
+    // Continuation where 'proceed' was called in order to properly walk over stacktrace
+    private var proceedContinuation: Continuation<*>? = null
     private var index = 0
 
     override fun finish() {
         index = blocks.size
     }
 
     override suspend fun proceed(): TSubject = suspendCoroutineUninterceptedOrReturn { continuation ->
+        proceedContinuation = continuation
```

Comment:
proceed()` could be invoked multiple times: this is why continuations are stacked in this case. Shouldn't `proceedContinuation` be restored back when leaving a proceed level?